src/main/java/tasklist/TaskList.java

```diff
@@ -0,0 +1,198 @@
+package tasklist;
+
+import static data.task.Task.containsDate;
+import static ui.Ui.echo;
+import static ui.Ui.line;
+import static ui.Ui.stringTaskAdded;
+import static ui.Ui.taskAdded;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+
+import data.task.Task;
+import storage.Storage;
+import ui.Ui;
+
+public class TaskList {
+    // Task storage
+    private ArrayList<Task> storage;
+    // Date storage
+    private HashMap<String, ArrayList<Task>> dateStorage;
+
+    /**
+     * constructor for TaskList
+     */
+    public TaskList() {
+        storage = new ArrayList<>();
+        dateStorage = new HashMap<>();
+    }
+
+    /**
+     * overloaded constructor that accepts Storage and updates list of tasks
+     * @param stores class that handles the save file
+     */
+    public TaskList(Storage stores) {
+        storage = new ArrayList<>();
+        dateStorage = new HashMap<>();
+        stores.load(this);
+    }
+
+    public ArrayList<Task> getStorage() {
+        return storage;
+    }
+
+    public HashMap<String, ArrayList<Task>> getDateStorage() {
+        return dateStorage;
+    }
+
+    /**
+     * Marks a certain task as complete
+     * @param taskNumber the index number of the task
+     */
+    public void markComplete(int taskNumber) {
+        Task currentTask = storage.get(taskNumber);
+        currentTask.complete();
+        line();
+        System.out.println("Marked task " + (taskNumber + 1) + " as complete.");
+        System.out.println(currentTask);
+        line();
+    }
+
+    /**
+     * String version of markComplete
+     * @param taskNumber index number of task
+     * @return string output of task that's marked complete
+     */
+    public String stringMarkComplete(int taskNumber) {
+        StringBuilder response = new StringBuilder();
+        Task currentTask = storage.get(taskNumber);
+        currentTask.complete();
+        response.append("Marked task ").append(taskNumber + 1).append(" as complete. \n");
+        response.append(currentTask);
+        return response.toString();
+    }
+
+
+    /**
+     * Adds a task to the task storage
+     * @param type the type of task to be added
+     * @param name the name of the task
+     */
+    public void store(String type, String name) {
+        Task taskToAdd = new Task(type, name);
+        storage.add(taskToAdd);
+        if (taskToAdd.hasDate()) {
+            if (!dateStorage.containsKey(taskToAdd.getDate())) {
+                dateStorage.put(taskToAdd.getDate(), new ArrayList<>());
+            }
+            dateStorage.get(taskToAdd.getDate()).add(taskToAdd);
+        }
+        taskAdded(taskToAdd, storage);
+    }
```

Comment:
Feel like this is the job of a storage class, the TaskList class should only be responsible to adding a new task to the tasklist, deleting task from tasklist etc.